src/boosting/gbdt.cpp

```diff
@@ -15,6 +15,12 @@
 #include <ctime>
 #include <sstream>
 
+#ifdef __APPLE__
+#include <sys/malloc.h>
+#else
+#include <malloc.h>
+#endif
```

Comment:
I'm really curious, why was this necessary? Is there something missing from LightGBM's existing checks, that should be added to LightGBM independent of this PR?

https://github.com/microsoft/LightGBM/blob/4ae3d1387de73fbadf6194faff816a3b446e506c/CMakeLists.txt#L287-LL300

https://github.com/microsoft/LightGBM/blob/b0137debe6e9cc92b65ec71b0fe8a56ea213c143/include/LightGBM/utils/common.h#L46-L60

Reply:
You are right. That should be deleted.